dspace-api/src/main/java/org/dspace/content/dao/impl/ItemDAOImpl.java

```diff
@@ -175,59 +179,62 @@ protected ItemDAOImpl() {
                 }
             }
 
-            DetachedCriteria subcriteria = DetachedCriteria.forClass(MetadataValue.class, "mv");
-            subcriteria.add(Property.forName("mv.dSpaceObject").eqProperty("item.id"));
-            subcriteria.setProjection(Projections.property("mv.dSpaceObject"));
+            Subquery<DSpaceObject> subquery = criteriaQuery.subquery(DSpaceObject.class);
+            Root<MetadataValue> subqueryRoot = subquery.from(MetadataValue.class);
+            subquery.select(subqueryRoot.get(MetadataValue_.dSpaceObject));
+            List<Predicate> predicateListSubQuery = new LinkedList<>();
 
-            if (!listFieldList.get(i).isEmpty()) {
-                subcriteria.add(Restrictions.in("metadataField", listFieldList.get(i)));
-            }
+            predicateListSubQuery.add(criteriaBuilder.equal(subqueryRoot.get(MetadataValue_.dSpaceObject), itemRoot
+                .get(Item_.id)));
 
-            sb.append(op.name() + " ");
-            if (op == OP.equals || op == OP.not_equals) {
-                subcriteria.add(Property.forName("mv.value").eq(query_val.get(i)));
-                sb.append(query_val.get(i));
-            } else if (op == OP.like || op == OP.not_like) {
-                subcriteria.add(Property.forName("mv.value").like(query_val.get(i)));
-                sb.append(query_val.get(i));
-            } else if (op == OP.contains || op == OP.doesnt_contain) {
-                subcriteria.add(Property.forName("mv.value").like("%" + query_val.get(i) + "%"));
-                sb.append(query_val.get(i));
-            } else if (op == OP.matches || op == OP.doesnt_match) {
-                subcriteria.add(Restrictions.sqlRestriction(regexClause, query_val.get(i), StandardBasicTypes.STRING));
-                sb.append(query_val.get(i));
+            if (!listFieldList.get(i).isEmpty()) {
+                predicateListSubQuery.add(criteriaBuilder.isTrue(subqueryRoot.get(MetadataValue_.metadataField).in
+                    (listFieldList.get(i))));
             }
-
-            if (op == OP.exists || op == OP.equals || op == OP.like || op == OP.contains || op == OP.matches) {
-                criteria.add(Subqueries.exists(subcriteria));
-            } else {
-                criteria.add(Subqueries.notExists(subcriteria));
+            if (op == OP.equals) {
```

Comment:
Why are we changing the structure of this `if` statement?  The old code has several areas that say `if(something || something else) { do the same thing }`.  It looks like here it's all broken out into several extra `else if` statements?

Reply:
The end result doesn't seem to be exactly the same as the original code...at least at a glance. But, maybe I'm missing something here?